src/frontends/pytorch/src/op/native_multi_head_attention.cpp

```diff
@@ -0,0 +1,192 @@
+    const auto qv_transpose_dims = context.mark_node(opset10::Constant::create(element::i64, Shape{4}, {0, 2, 1, 3}));
+    const auto k_transpose_dims = context.mark_node(opset10::Constant::create(element::i64, Shape{4}, {0, 2, 3, 1}));
+
+    const auto query_reshaped =
+        context.mark_node(std::make_shared<opset10::Reshape>(query_biased, qkv_reshape_dims, false));
+    const auto key_reshaped =
+        context.mark_node(std::make_shared<opset10::Reshape>(key_biased, qkv_reshape_dims, false));
+    const auto value_reshaped =
+        context.mark_node(std::make_shared<opset10::Reshape>(value_biased, qkv_reshape_dims, false));
+
+    const auto query_transposed =
+        context.mark_node(std::make_shared<opset10::Transpose>(query_reshaped, qv_transpose_dims));
+    const auto key_transposed = context.mark_node(std::make_shared<opset10::Transpose>(key_reshaped, k_transpose_dims));
+    const auto value_transposed =
+        context.mark_node(std::make_shared<opset10::Transpose>(value_reshaped, qv_transpose_dims));
+
+    const auto scale_one = context.mark_node(std::make_shared<opset10::ConvertLike>(one_1d, query_transposed));
+    const auto scale_dim = context.mark_node(std::make_shared<opset10::ConvertLike>(embed_div_heads, query_transposed));
+    const auto scale_dim_sqrt = context.mark_node(std::make_shared<opset10::Sqrt>(scale_dim));
+    const auto scale = context.mark_node(std::make_shared<opset10::Divide>(scale_one, scale_dim_sqrt));
+
+    const auto query_key_transpose_dot_product =
+        context.mark_node(std::make_shared<opset10::MatMul>(query_transposed, key_transposed));
+
+    auto scaled_dot_product =
+        context.mark_node(std::make_shared<opset10::Multiply>(query_key_transpose_dot_product, scale));
+
+    // Mask handling
+    if (!context.input_is_none(12)) {
+        auto atten_mask = context.get_input(9);
+        if (atten_mask.get_element_type() == element::boolean) {
+            const auto minus_inf_conv =
+                context.mark_node(std::make_shared<opset10::ConvertLike>(minus_inf, scaled_dot_product));
+            const auto mask_inverse = context.mark_node(std::make_shared<opset10::LogicalNot>(atten_mask));
+            atten_mask = context.mark_node(std::make_shared<opset10::ConvertLike>(atten_mask, scaled_dot_product));
+            atten_mask = context.mark_node(std::make_shared<opset10::Select>(mask_inverse, atten_mask, minus_inf_conv));
+        } else {
+            // Once int/float mask type is supported in PyTorch,
+            // remove this assert to allow for such masks in OV
+            FRONT_END_OP_CONVERSION_CHECK(1, "Non-boolean masks are not supported.");
+            atten_mask = context.mark_node(std::make_shared<opset10::ConvertLike>(atten_mask, scaled_dot_product));
+        }
+
+        if (context.const_input<int64_t>(12) == 1) {
+            const auto target_mask_reshape = context.mark_node(
```

Comment:
Could you add comment why you reshape only if `mask_type` == 1?

Reply:
Added! Additionally I removed the unnecessary broadcast if mask type == 2.